app/controllers/coronavirus_form/address_lookup_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class CoronavirusForm::AddressLookupController < ApplicationController
+  def show
+    @addresses = helpers.postcode_lookup(session[:postcode])
+    render controller_path, status: :ok
+  rescue AddressLookupError, AddressNotFoundError => e
+    session[:flash] = e.message
+
+    if e.class.name == "AddressLookupError" && e.status == 401
```

Comment:
I would recommend raising a different error in the helper for the authentication error (e.g. `AddressAuthError`), then you can have a simpler controller, e.g.

```ruby
rescue AddressAuthError
      GovukError.notify("API Key is invalid!")
      redirect_to support_address_path
rescue AddressLookupError, AddressNotFoundError => e
      session[:flash] = e.message
      redirect_to postcode_lookup_url
```

Reply:
Nice shout, have just pushed this change up :)